File: FashionClothesAndTrends.Domain/Entities/OrderAggregate/OrderHistory.cs
using FashionClothesAndTrends.Domain.Common;
using FashionClothesAndTrends.Domain.Entities.Enums;

namespace FashionClothesAndTrends.Domain.Entities.OrderAggregate;

public class OrderHistory : BaseEntity
{
    public DateTime OrderDate { get; set; }
    public decimal TotalAmount { get; set; }
    public OrderStatus Status { get; set; }
    public string ShippingAddress { get; set; }
    public ICollection<OrderItemHistory> OrderItems { get; set; }
    public string UserId { get; set; }
    public User User { get; set; }
}

File: ClientApp/src/app/shared/models/order-history.ts

export interface OrderHistory {
  id: string;
  orderDate: Date;
  totalAmount: number;
  status: string;
  shippingAddress: string;
  orderItemsHisory: OrderItemHistory[];
}

export interface OrderItemHistory {
  id: string;
  clothingItemId: string;
  clothingItemName: string;
  quantity: number;
  priceAtPurchase: number;
}
